File: java/src/main/java/fr/sdecout/handson/rest/book/BookUpdate.java
package fr.sdecout.handson.rest.book;

import fr.sdecout.handson.rest.author.AuthorId;

import java.util.Collection;
import java.util.stream.Stream;

public interface BookUpdate {
    void save(Isbn isbn, String title, Collection<AuthorId> authors);
}

File: kotlin/src/main/kotlin/fr/sdecout/handson/rest/book/BookUpdate.kt
package fr.sdecout.handson.rest.book

import fr.sdecout.handson.rest.author.AuthorId
import fr.sdecout.handson.rest.shared.Isbn

fun interface BookUpdate {
    fun save(isbn: Isbn, title: String, authors: Collection<AuthorId>)
}
